examples/snmp/pfsense.js

```diff
@@ -0,0 +1,1144 @@
+/**
+ * This driver extracts information about the network, cpu and some system information for pfsense
+ * The communication protocol is SNMP and SSH
+ * This driver create a dynamic monitoring variables specified in the variable {@link config_paramters}
+ * Tested under pfsense 2.6.0-RELEASE
+ */
+
+var createSNMPSession = D.device.createSNMPSession;
+var _var = D.device.createVariable;
+
+/** ssh init */
+
+var ssh_config = {
+    username: D.device.username(),
+    password: D.device.password(),
+    port: 22,
+    timeout: 30000
+};
+
+var ntp_status_cmd = "ntpq -p | awk '{print $1 \";\" $2 \";\" $7}'";
+var dns_servers_cmd = "grep \"nameserver\" /etc/resolv.conf | awk '{system(\"dig +tries=1 @\" $2 \" google.com > /dev/null; echo \" $2 \":$?\")}'";
+
+function exec_command(last, next) {
+    var config = clone({ command: this.command }, ssh_config);
+    D.device.sendSSHCommand(config, function (out, err) {
```

Comment:
what if the ssh command fails?

Reply:
if the ssh fails no data will be passed to the next function, so the variable will be ignored